File: 01/calo.pl
#!/usr/bin/perl

my $input_file = './input';

open(my $input_data, '<', $input_file);

my $current_sum = 0;
my $max = 0;

my @sums;
my $sum_idx = 0;

while(<$input_data>) {
    chomp $_;

    if($_ ne "") {
        $current_sum += int($_);
        next if not eof $input_data;
    }

    $sums[$sum_idx++] = $current_sum;
    $max = $current_sum if $current_sum > $max;
    $current_sum = 0;
}

@sorted_sums = sort { $b <=> $a } @sums;

my $idx = 0;
my $top_sum = 0;

while($idx < 3) {
    $top_sum += $sorted_sums[$idx];
    $idx++;
}

print "Answer for part 1 is $max\n";
print "Answer for part 2 is $top_sum\n";

File: 01/calo.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <bits/stdc++.h>

int main(void) {
    static const char* input_file = "./input";

    std::ifstream input_data(input_file);
    std::string line;

    long sum = 0;

    std::vector<long> sums;

    while(std::getline(input_data, line)) {
        if(line != "") {
            sum += std::stol(line);
            if(input_data.peek() != EOF) continue;
        }

        sums.push_back(sum);
        sum = 0;
    }

    std::sort(sums.begin(), sums.end());

    std::cout << "Anwser for part 1 is " << sums[sums.size() - 1] << '\n';
    std::cout << "Anwser for part 1 is " <<
        sums[sums.size() - 1] + sums[sums.size() - 2] + sums[sums.size() - 3] << '\n';

    return 0;
}
